core-kotlin-modules/core-kotlin-advanced-2/src/main/kotlin/com/baeldung/generic/inline/Samples.kt

```diff
@@ -0,0 +1,46 @@
+package com.baeldung.generic.inline
+
+@JvmInline
+value class Wrapper<T>(val value: T)
+
+fun main() {
+    val wrapper = Wrapper<Int>(42)
+    val value: Int = wrapper.value // No runtime overhead
+    val list = listOf(42)
+    val item: Int = list[0] // Boxing and unboxing overhead
+}
+
+@JvmInline
+value class Id<T>(val value: T)
+
+interface Identifiable<T> {
+    val id: Id<T>
+}
+
+data class User(override val id: Id<String>, val name: String) : Identifiable<String>
+
+data class Wallet(override val id: Id<Int>, val currency: String) : Identifiable<Int>
+
+fun <T : Identifiable<*>> findById(entities: List<T>, id: Id<*>): T? {
+    return entities.firstOrNull { it.id == id }
+}
```

Comment:
Thinking out loud here, would this read better?

```suggestion
fun <T : Identifiable<ID: Any>> findById(entities: List<T>, id: Id<ID>): T? {
    return entities.firstOrNull { it.id == id }
}
```

Reply:
I think it is clearer, yes!
In production I would probably leave the wildcard to mean "I'm leaving this as I intend to not care about the type". For tutorial reasons (and for this, specially) we totally care about the type xD